Mathlib/Analysis/Convex/Topology.lean

```diff
@@ -315,6 +268,64 @@ protected theorem Convex.strictConvex {s : Set E} (hs : Convex 𝕜 s)
 
 end ContinuousConstSMul
 
+section TopologicalSpace
+
+variable [OrderedSemiring 𝕜] [AddCommGroup E] [Module 𝕜 E] [TopologicalSpace E]
+
+theorem convex_closed_sInter {S : Set (Set E)} (h : ∀ s ∈ S, Convex 𝕜 s ∧ IsClosed s) :
+    Convex 𝕜 (⋂₀ S) ∧ IsClosed (⋂₀ S) :=
+  ⟨fun _ hx => starConvex_sInter fun _ hs => (h _ hs).1 <| hx _ hs,
+    isClosed_sInter fun _ hs => (h _ hs).2⟩
+
+variable (𝕜)
+
+/-- The convex closed hull of a set `s` is the minimal convex closed set that includes `s`. -/
+@[simps! isClosed]
+def closedConvexHull : ClosureOperator (Set E) := .ofCompletePred (fun s => Convex 𝕜 s ∧ IsClosed s)
+  fun _ ↦ convex_closed_sInter
+
+variable {𝕜}
+
+theorem convex_closedConvexHull {s : Set E} :
+    Convex 𝕜 (closedConvexHull 𝕜 s) := ((closedConvexHull 𝕜).isClosed_closure s).1
+
+theorem isClosed_closedConvexHull {s : Set E} :
+    IsClosed (closedConvexHull 𝕜 s) := ((closedConvexHull 𝕜).isClosed_closure s).2
+
+theorem subset_closedConvexHull {s : Set E} : s ⊆ closedConvexHull 𝕜 s :=
+  (closedConvexHull 𝕜).le_closure s
+
+theorem closure_subset_closedConvexHull {s : Set E} : closure s ⊆ closedConvexHull 𝕜 s :=
+  closure_minimal subset_closedConvexHull isClosed_closedConvexHull
+
+theorem closedConvexHull_min {s t : Set E} (hst : s ⊆ t) (h_conv : Convex 𝕜 t)
+    (h_closed : IsClosed t) : closedConvexHull 𝕜 s ⊆ t :=
+  (closedConvexHull 𝕜).closure_min hst ⟨h_conv, h_closed⟩
+
+theorem convexHull_subseteq_closedConvexHull {s : Set E} :
+    (convexHull 𝕜) s ⊆ (closedConvexHull 𝕜) s :=
+  convexHull_min subset_closedConvexHull convex_closedConvexHull
+
+theorem closedConvexHull_eq_closedConvexHull_closure {s : Set E} :
+    closedConvexHull 𝕜 s = closedConvexHull 𝕜 (closure s) :=
```

Comment:
Now that I look at this again, I wonder if it makes more sense turned around as a `simp` lemma?

Reply:
Done, thanks.